File: template/app/src/main/java/com/example/base/baseapp/repository/local/ExampleDataLocalRepository.java
package com.example.base.baseapp.repository.local;

import com.example.base.baseapp.dagger.scope.AppScope;
import com.example.base.baseapp.model.ExampleData;
import com.example.base.baseapp.persistence.ExampleDataDatabase;
import com.example.base.baseapp.repository.ExampleDataRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Completable;
import io.reactivex.Single;

@AppScope
public class ExampleDataLocalRepository implements ExampleDataRepository {

    private ExampleDataDatabase exampleDataDatabase;

    @Inject
    public ExampleDataLocalRepository(ExampleDataDatabase exampleDataDatabase) {
        this.exampleDataDatabase = exampleDataDatabase;
    }

    @Override
    public Single<List<ExampleData>> getAllExampleData() {
        return exampleDataDatabase.exampleDataDao().getAllExampleData();
    }

    @Override
    public Completable insertExampleData(List<ExampleData> exampleDataList) {
        return Completable.fromAction(() -> exampleDataDatabase.exampleDataDao().insertExampleData(exampleDataList));
    }
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/repository/local/ExampleDataLocalRepository.kt
package com.example.base.baseapp.repository.local

import com.example.base.baseapp.model.ExampleData
import com.example.base.baseapp.persistence.ExampleDataDatabase
import com.example.base.baseapp.repository.ExampleDataRepository

import javax.inject.Inject

import io.reactivex.Completable
import io.reactivex.Single

class ExampleDataLocalRepository @Inject
constructor(private val exampleDataDatabase: ExampleDataDatabase) : ExampleDataRepository {

    override val allExampleData: Single<List<ExampleData>>
        get() = exampleDataDatabase.exampleDataDao().allExampleData

    override fun insertExampleData(exampleDataList: List<ExampleData>): Completable {
        return Completable.fromAction { exampleDataDatabase.exampleDataDao().insertExampleData(exampleDataList) }
    }
}
